compiler/parser.mly

```diff
@@ -138,16 +138,16 @@ stmt_list:
 
 stmt:
     expr ENDLINE                                { Expr($1) } 
-  | FOR LPAREN expr COMMA bool_expr COMMA
-    assignment_stmt RPAREN
-    LCURLY stmt_list RCURLY ENDLINE             { For($3, $5, $7, $10) } 
-  | WHILE LPAREN bool_expr RPAREN
-    LCURLY stmt_list RCURLY ENDLINE             { While($3, $6) }
+  | FOR LPAREN assignment_stmt COMMA bool_expr COMMA
```

Comment:
Assignment statement in the parser include `ID ASSIGN expr { Update_variable($1, $3) }` (line 179 of parser.mly)

So this actually passes:

```
int a = 1
for (a = 2 , a < 4, a = a + 1) {
print("swer")
}
```
Is that what you were referring to?